libs/mpeg-transport-stream/src/main/java/org/codice/ddf/libs/mpeg/transport/MpegTransportStreamMetadataExtractor.java

```diff
@@ -106,17 +111,29 @@ private void extractTransportStreamMetadata(final BiConsumer<Integer, byte[]> ca
 
         source.reset();
 
-        MTSPacket transportStreamPacket;
+        MTSValidPacketIterator packetIterator = new MTSValidPacketIterator(source);
+        MTSPacket transportStreamPacket = packetIterator.getNextValidPacket();
 
-        while ((transportStreamPacket = source.nextPacket()) != null) {
-            final int packetId = transportStreamPacket.getPid();
+        try {
+            if (transportStreamPacket != null) {
+                LOGGER.debug("First valid packet found after {} failures",
```

Comment:
@stustison said this log (and the one below it) can remain debug since they are only executed once.